src/core/cuda_utils.cc

```diff
@@ -87,7 +96,14 @@ CopyBuffer(
   // the src buffer is valid.
   if ((src_memory_type != TRITONSERVER_MEMORY_GPU) &&
       (dst_memory_type != TRITONSERVER_MEMORY_GPU)) {
-    memcpy(dst, src, byte_size);
+    if (zero_copy_support) {
+      auto params = new CopyParams(dst, src, byte_size);
+      cudaLaunchHostFunc(
+          cuda_stream, MemcpyHost, reinterpret_cast<void*>(params));
+      *cuda_used = true;
```

Comment:
This should take care of the synchronization at the end with Finalize() call.. Hence, I don't think we need any other synchronization steps.

Reply:
Yes that is correct. I have remove the extra synchronization I had added.